src/app/day-calendar/day-calendar.component.ts

```diff
@@ -188,6 +188,21 @@ export class DayCalendarComponent implements OnInit, OnChanges, ControlValueAcce
     return this.dayCalendarService.getDayBtnText(this.componentConfig, day.date);
   }
 
+  getDayBtnCssClass(day: IDay): {[klass: string]: any} {
+    const cssClasses: {[klass: string]: any} = {
```

Comment:
why the `{[klass: string]: any}` uses `any`, isn't it always a boolean?

Reply:
I used the definition that angular does for ngClass: https://angular.io/api/common/NgClass

It's probably because ngClass only needs it to evaluate to something truthy.

Reply:
let's do it with booleans, it is less generic